### ISSUE TITLE
Import QuaggaJS barcode-scanner

### ISSUE THREAD

**Issue by NightWnvol** [ionic-team/stencil#452]
**Stencil version:**

```
 @stencil/core@0.2.3
```

**I'm submitting a:**

- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or https://stencil-worldwide.slack.com

**Current behavior:**

Build failed:

```
[37:06.5]  changed file: app-home.tsx
[37:06.5]  rebuild, app, dev mode, started ...
[37:06.5]  compile started ...
[37:09.0]  compile finished in 2.51 s
preferring built-in module 'punycode' over local alternative at 'C:\Users\NightW\Desktop\test-stencil\my-app\node_modules\punycode\punycode.js', pass 'preferBuiltins: false' to disable this behavior or 'preferBuiltins: true' to disable this warning
preferring built-in module 'punycode' over local alternative at 'C:\Users\NightW\Desktop\test-stencil\my-app\node_modules\punycode\punycode.js', pass 'preferBuiltins: false' to disable this behavior or 'preferBuiltins: true' to disable this warning
preferring built-in module 'punycode' over local alternative at 'C:\Users\NightW\Desktop\test-stencil\my-app\node_modules\punycode\punycode.js', pass 'preferBuiltins: false' to disable this behavior or 'preferBuiltins: true' to disable this warning
preferring built-in module 'punycode' over local alternative at 'C:\Users\NightW\Desktop\test-stencil\my-app\node_modules\punycode\punycode.js', pass 'preferBuiltins: false' to disable this behavior or 'preferBuiltins: true' to disable this warning
preferring built-in module 'punycode' over local alternative at 'C:\Users\NightW\Desktop\test-stencil\my-app\node_modules\punycode\punycode.js', pass 'preferBuiltins: false' to disable this behavior or 'preferBuiltins: true' to disable this warning

[ WARN  ]  app-home, my-app: 'path' is imported by node_modules\get-pixels\node-pixels.js, but could not be resolved – treating it as an external dependency

[ WARN  ]  app-home, my-app: 'util' is imported by node_modules\pngjs\lib\png.js, but could not be resolved – treating it as an external dependency

[ WARN  ]  app-home, my-app: 'stream' is imported by node_modules\pngjs\lib\png.js, but could not be resolved – treating it as an external dependency

[ WARN  ]  app-home, my-app: 'util' is imported by node_modules\pngjs\lib\parser-async.js, but could not be resolved – treating it as an external dependency

[ WARN  ]  app-home, my-app: 'zlib' is imported by node_modules\pngjs\lib\parser-async.js, but could not be resolved – treating it as an external dependency

[ WARN  ]  app-home, my-app: 'util' is imported by node_modules\pngjs\lib\chunkstream.js, but could not be resolved – treating it as an external dependency

[ WARN  ]  app-home, my-app: 'stream' is imported by node_modules\pngjs\lib\chunkstream.js, but could not be resolved – treating it as an external dependency

[ WARN  ]  app-home, my-app: 'util' is imported by  commonjs-external:util, but could not be resolved – treating it as an external dependency

[ WARN  ]  app-home, my-app: 'stream' is imported by  commonjs-external:stream, but could not be resolved – treating it as an external dependency

[ WARN  ]  app-home, my-app: 'util' is imported by node_modules\pngjs\lib\filter-parse-async.js, but could not be resolved – treating it as an external dependency

[ WARN  ]  app-home, my-app: 'zlib' is imported by  commonjs-external:zlib, but could not be resolved – treating it as an external dependency

[ WARN  ]  app-home, my-app: 'util' is imported by node_modules\pngjs\lib\packer-async.js, but could not be resolved – treating it as an external dependency

[ WARN  ]  app-home, my-app: 'stream' is imported by node_modules\pngjs\lib\packer-async.js, but could not be resolved – treating it as an external dependency

[ WARN  ]  app-home, my-app: 'zlib' is imported by node_modules\pngjs\lib\packer.js, but could not be resolved – treating it as an external dependency

[ WARN  ]  app-home, my-app: 'zlib' is imported by node_modules\pngjs\lib\parser-sync.js, but could not be resolved – treating it as an external dependency

[37:10.3]  rebuild failed, watching for changes... in 3.77 s
```

**Expected behavior:**

I should be able to import QuaggaJS barcode-scanner and use it

**Steps to reproduce:**

In `stancil-app-starter` project folder:
`npm install quagga`

In `app-home.tsx`
```
import { Component } from '@stencil/core';
import * as Quagga from 'quagga';

@Component({
  tag: 'app-home',
  styleUrl: 'app-home.scss'
})
export class AppHome {
  config = {
    inputStream: {
      name: 'Live',
      type: 'LiveStream'
    },
    frequency: 5,
    decoder: {
      readers: [
        'ean_reader',
        'ean_8_reader'
      ],
      debug: {
        showCanvas: true,
        showPatches: true,
        showFoundPatches: true,
        showSkeleton: true,
        showLabels: true,
        showPatchLabels: true,
        showRemainingPatchLabels: true,
        boxFromPatches: {
          showTransformed: true,
          showTransformedBox: true,
          showBB: true
        }
      }
    },
    locator: {
      patchSize: 'x-large'
    }
  };

  scan() {
    Quagga.init(this.config, error => {
      if (error) {
        alert(error);
        return;
      }
      Quagga.start();
    });
  }

  render() {
    return (
      <div>
        <p>
          Welcome to the Stencil App Starter.
          You can use this starter to build entire apps all with
          web components using Stencil!
          Check out our docs on <a href='https://stenciljs.com'>stenciljs.com</a> to get started.
        </p>
        <stencil-route-link url='/profile/stencil'>
          <button>
            Profile page
          </button>
        </stencil-route-link>
        <div>
          <button onClick={() => this.scan()}>Scan</button>
          <div id="interactive" class="viewport"></div>
        </div>
      </div>
    );
  }
}
```

**Related code:**

```tsx
// insert any relevant code here
```

**Other information:**

**Comment by bfmatei** [comment]
Hello,

As a temporary workaround you can replace the import of Quagga with
`import Quagga from 'quagga/dist/quagga.min';`

**Comment by jgw96** [comment]
Hello, thanks for opening an issue with us! Note to self, test this today with the new code splitting implementation as that should resolve this.

**Comment by jgw96** [comment]
@jthoms1 Just a note: I re-tested this with 0.4.0 and I am still able to reproduce with the exact error mentioned above.

**Comment by jthoms1** [comment]
I believe this was fixed with commit 256e70a8d3fff2b25ce884cbda2725630d03e7c1. So `@stencil/core@0.5.0-3` should contain the fix.

**Comment by jgw96** [comment]
Ill install and test

**Comment by jthoms1** [comment]
This issue is fixed with commit b1482390f03ab768ddb229a015f528927a64a3ae.
